var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -79,7 +79,8 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
 
     # Enable builds with the NVIDIA compiler
     patch('nvhpc-5.30.patch', when='@5.30.0:5.30.99 %nvhpc')
-    conflicts('@5.32.0:', when='%nvhpc',
+    patch('nvhpc-5.32.patch', when='@5.32.0:5.32.99 %nvhpc')
```

Comment:
Same comment as for the other patch

Reply:
The Nvidia compiler tries to be a drop in replacement for gcc, but there are some behaviors are not (yet) compatible.  The Perl Configure script identifies nvc as gcc and then tries to use one of these incompatible behaviors.  So the first change just prevents the Configure script from triggering that behavior.

The Makefile.SH change alters the install behavior to use the standalone "mini" Perl to install the "real" Perl, rather than the real Perl bootstrapping its own installation.  IIRC, at that time in the install the environment is not fully configured so the real Perl is not fully functional.

Reply:
Same as https://github.com/spack/spack/pull/21975#discussion_r607549949

Reply:
Added